infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/LogReplicationConfig.java

```diff
@@ -91,17 +98,81 @@ public LogReplicationConfig(Set<String> streamsToReplicate, int maxNumMsgPerBatc
     /**
      * Constructor
      *
-     * @param streamsToReplicate Unique identifiers for all streams to be replicated across sites.
+     * @param streamsToReplicate Info for all streams to be replicated across sites
      * @param maxNumMsgPerBatch snapshot sync batch size (number of entries per batch)
      */
-    public LogReplicationConfig(Set<String> streamsToReplicate, int maxNumMsgPerBatch, int maxMsgSize) {
-        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, MAX_CACHE_NUM_ENTRIES);
+    public LogReplicationConfig(Set<TableInfo> streamsToReplicate, int maxNumMsgPerBatch, int maxMsgSize) {
+        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, MAX_CACHE_NUM_ENTRIES, null);
     }
 
-    public LogReplicationConfig(Set<String> streamsToReplicate, Map<UUID, List<UUID>> streamingTagsMap,
-                                Set<UUID> mergeOnlyStreams, int maxNumMsgPerBatch, int maxMsgSize, int cacheSize) {
-        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, cacheSize);
+    public LogReplicationConfig(Set<TableInfo> streamsToReplicate, Map<UUID, List<UUID>> streamingTagsMap,
+                                Set<UUID> mergeOnlyStreams, int maxNumMsgPerBatch, int maxMsgSize,
+                                int cacheSize, LogReplicationStreamNameTableManager streamInfoManager) {
+        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, cacheSize, streamInfoManager);
         this.dataStreamToTagsMap = streamingTagsMap;
         this.mergeOnlyStreams = mergeOnlyStreams;
     }
-}
\ No newline at end of file
+
+    /**
+     * This class represents Log Replication streams info
+     */
+    public static class StreamInfo {
+        private final Set<UUID> streamIds;
+        private final LogReplicationStreamNameTableManager streamInfoManager;
+
+        public StreamInfo(@NonNull Set<TableInfo> streamsToReplicate,
+                          LogReplicationStreamNameTableManager streamInfoManager) {
+            this.streamInfoManager = streamInfoManager;
+            this.streamIds = new HashSet<>();
+            refreshStreamIds(streamsToReplicate, true);
+        }
+
+        private void refreshStreamIds(Set<TableInfo> streamsToReplicate, boolean shouldClear) {
+            if (shouldClear) {
+                this.streamIds.clear();
+            }
+
+            for (TableInfo info : streamsToReplicate) {
+                if (info.hasField(TableInfo.getDescriptor().findFieldByName("name"))) {
+                    this.streamIds.add(CorfuRuntime.getStreamID(info.getName()));
+                } else if (info.hasField(TableInfo.getDescriptor().findFieldByName("id"))) {
+                    this.streamIds.add(UUID.fromString(info.getId()));
```

Comment:
I thought you were meaning "id" is not set? in this case I thought it will return false, as protobuf 3 now are having default values. Nan previously was using `== null` here which would always return true, so I changed to `hasField` which detects whether the field is set or not